src/components/List/List.tsx

```diff
@@ -1,27 +1,38 @@
 import * as React from 'react';
-import PropTypes from 'prop-types';
 import bem from '../../utils/bem';
-import ListItem from './ListItem';
 import styles from './List.scss';
 import { LIST_NAVIGATION_DIRECTIONS, ENTER_KEY } from '../../constants';
 import scrollIntoViewIfNeeded from '../../utils/scrollIntoViewIfNeeded';
 
+interface Props extends React.HTMLAttributes<HTMLUListElement> {
+    /** List items. They should be ListItem or li.
+     * If you want to add an element that is not part of the list, but should be rendered within it
+     * (e.g. an image to decorate the list) use `data-list-exception` as a prop to that child
+     */
+    children: React.ReactElement[] | null;
+    /** Adds dividing lines between the list items */
+    isDivided?: boolean;
+    /** Defines if selection should be made on navigate */
+    doSelectOnNavigate?: boolean;
+    /** manage keyboard navigation externally */
+    isControlledNavigation?: boolean;
+    /** Ref to access the li element */
+    ref?: React.RefObject<HTMLUListElement>;
+}
+
 const { block, elem } = bem('List', styles);
 
-const isNotListItem = element => element && element.type !== ListItem && element.type !== 'li';
+export const NOT_LIST_CHILD = 'data-list-exception';
 const NAVIGATION_STEP_VALUES = {
     [LIST_NAVIGATION_DIRECTIONS.UP]: -1,
     [LIST_NAVIGATION_DIRECTIONS.DOWN]: 1,
 };
 
-export const NOT_LIST_CHILD = 'data-list-exception';
-export const LIST_CHILD = 'data-list-child';
-
-const List = React.forwardRef((props, ref) => {
+const List: React.FC<Props> = React.forwardRef((props, ref) => {
```

Comment:
`React.forwardRef<T, P = {}>` is a generic function which accepts 2 arguments:

* T is type of our DOM element
* P is type of our props

That gives a better type of `ref.current` value in the end.

So here it can be:
```ts
const List: React.FC<Props> = React.forwardRef<HTMLUListElement, Props>((props, ref) => {
```

Reply:
I think we tried to use that standard, but had some problems with it. Hence we went with defining the `ref` type in the interface. This approach also has the advantage that it shows up in storybook, so it becomes more transparent what the component expects and accepts

Reply:
ok then